src/components/NavBar/NavBar.spec.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+
+import NavBar from './NavBar.component';
+import DarThemeProvider from '../../providers/DarkMode';
+import AuthProvider from '../../providers/Auth';
+import { storage } from '../../utils/storage';
+
+jest.mock('../../utils/storage');
+
+describe('Testing NavBar.component', () => {
+  test('toggleDarkmode when click toggle button', () => {
+    const { container } = render(
+      <AuthProvider>
+        <DarThemeProvider>
+          <NavBar />
+        </DarThemeProvider>
+      </AuthProvider>
+    );
+
+    const darkModeToggle = container.querySelector('#toggleTheme');
+    fireEvent.click(darkModeToggle);
+
+    expect(storage.set).toHaveBeenCalledTimes(1);
+    fireEvent.click(darkModeToggle);
+    expect(storage.set).toHaveBeenCalledTimes(2);
```

Comment:
Nice testing, since `storage.set` is a mock, you could also check what arguments it was called with, this way you have a more resilient unit test. Something like:

```javascript
expect(storage.set).toHaveBeenNthCalledWith(N, something1, somethingN);
```

Where N is the number of the call, and something1...somethingN the arguments passed to that function.